<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
  "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns='http://www.w3.org/1999/xhtml' xml:lang='en' lang='en'>
<head>
<title>svntogit/packages.git - Git clone of the 'packages' repository
</title>
<meta name='generator' content='cgit v0.10.2'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='shortcut icon' href='/favicon.ico'/>
<link rel='alternate' title='Atom feed' href='https://projects.archlinux.org/svntogit/packages.git/atom/trunk/git-fixes.patch?h=packages/xf86-video-intel' type='application/atom+xml'/>
<link rel='vcs-git' href='git://projects.archlinux.org/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
<link rel='vcs-git' href='http://projects.archlinux.org/git/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
<link rel='vcs-git' href='https://projects.archlinux.org/git/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
<link rel='vcs-git' href='ssh://gerolde.archlinux.org/srv/projects/git/svntogit/packages.git' title='svntogit/packages.git Git repository'/>
</head>
<body>
	<div id="archnavbar"><!-- Arch Linux global navigation bar -->
		<div id="archnavbarlogo">
			<p><a href="http://www.archlinux.org/" title="Arch news, packages, projects and more"></a></p>
		</div>
		<div id="archnavbarmenu">
			<ul id="archnavbarlist">
				<li id="anb-home"><a href="http://www.archlinux.org/" title="Arch news, packages, projects and more">Home</a></li>
				<li id="anb-packages"><a href="http://www.archlinux.org/packages/" title="Arch Package Database">Packages</a></li>
				<li id="anb-forums"><a href="https://bbs.archlinux.org/" title="Community forums">Forums</a></li>
				<li id="anb-wiki"><a href="https://wiki.archlinux.org/" title="Community documentation">Wiki</a></li>
				<li id="anb-bugs"><a href="https://bugs.archlinux.org/" title="Report and follow bugs">Bugs</a></li>
				<li id="anb-aur"><a href="https://aur.archlinux.org/" title="Arch Linux User Repository">AUR</a></li>
				<li id="anb-download"><a href="http://www.archlinux.org/download/" title="Get Arch Linux">Download</a></li>
			</ul>
		</div>
	</div><!-- #archnavbar -->
<div id='cgit'><table id='header'>
<tr>
<td class='main'><a href='/'>index</a> : <a title='svntogit/packages.git' href='/svntogit/packages.git/'>svntogit/packages.git</a></td></tr>
<tr><td class='sub'>Git clone of the 'packages' repository
</td><td class='sub right'></td></tr></table>
<table class='tabs'><tr><td>
<a href='/svntogit/packages.git/?h=packages/xf86-video-intel'>summary</a><a href='/svntogit/packages.git/refs/?h=packages/xf86-video-intel'>refs</a><a href='/svntogit/packages.git/log/trunk/git-fixes.patch?h=packages/xf86-video-intel'>log</a><a class='active' href='/svntogit/packages.git/tree/trunk/git-fixes.patch?h=packages/xf86-video-intel'>tree</a><a href='/svntogit/packages.git/commit/trunk/git-fixes.patch?h=packages/xf86-video-intel'>commit</a><a href='/svntogit/packages.git/diff/trunk/git-fixes.patch?h=packages/xf86-video-intel'>diff</a><a href='/svntogit/packages.git/stats/trunk/git-fixes.patch?h=packages/xf86-video-intel'>stats</a></td><td class='form'><form class='right' method='get' action='/svntogit/packages.git/log/trunk/git-fixes.patch'>
<input type='hidden' name='h' value='packages/xf86-video-intel'/><select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/svntogit/packages.git/tree/?h=packages/xf86-video-intel'>root</a>/<a href='/svntogit/packages.git/tree/trunk?h=packages/xf86-video-intel'>trunk</a>/<a href='/svntogit/packages.git/tree/trunk/git-fixes.patch?h=packages/xf86-video-intel'>git-fixes.patch</a></div><div class='content'>blob: 20a0167c3cda4e4f6ae218481d5b597193a3da8c (<a href='/svntogit/packages.git/plain/trunk/git-fixes.patch?h=packages/xf86-video-intel'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
<a id='n37' href='#n37'>37</a>
<a id='n38' href='#n38'>38</a>
<a id='n39' href='#n39'>39</a>
<a id='n40' href='#n40'>40</a>
<a id='n41' href='#n41'>41</a>
<a id='n42' href='#n42'>42</a>
<a id='n43' href='#n43'>43</a>
<a id='n44' href='#n44'>44</a>
<a id='n45' href='#n45'>45</a>
<a id='n46' href='#n46'>46</a>
<a id='n47' href='#n47'>47</a>
<a id='n48' href='#n48'>48</a>
<a id='n49' href='#n49'>49</a>
<a id='n50' href='#n50'>50</a>
<a id='n51' href='#n51'>51</a>
<a id='n52' href='#n52'>52</a>
<a id='n53' href='#n53'>53</a>
<a id='n54' href='#n54'>54</a>
<a id='n55' href='#n55'>55</a>
<a id='n56' href='#n56'>56</a>
<a id='n57' href='#n57'>57</a>
<a id='n58' href='#n58'>58</a>
<a id='n59' href='#n59'>59</a>
<a id='n60' href='#n60'>60</a>
<a id='n61' href='#n61'>61</a>
<a id='n62' href='#n62'>62</a>
<a id='n63' href='#n63'>63</a>
<a id='n64' href='#n64'>64</a>
<a id='n65' href='#n65'>65</a>
<a id='n66' href='#n66'>66</a>
<a id='n67' href='#n67'>67</a>
<a id='n68' href='#n68'>68</a>
<a id='n69' href='#n69'>69</a>
<a id='n70' href='#n70'>70</a>
<a id='n71' href='#n71'>71</a>
<a id='n72' href='#n72'>72</a>
<a id='n73' href='#n73'>73</a>
<a id='n74' href='#n74'>74</a>
<a id='n75' href='#n75'>75</a>
<a id='n76' href='#n76'>76</a>
<a id='n77' href='#n77'>77</a>
<a id='n78' href='#n78'>78</a>
<a id='n79' href='#n79'>79</a>
<a id='n80' href='#n80'>80</a>
<a id='n81' href='#n81'>81</a>
<a id='n82' href='#n82'>82</a>
</pre></td>
<td class='lines'><pre><code>From 127aae5a72a69df325fed0b63d345e81583d5ca1 Mon Sep 17 00:00:00 2001
From: Chris Wilson &lt;chris&#64;chris-wilson.co.uk&gt;
Date: Mon, 16 Feb 2015 21:37:35 +0000
Subject: tools/intel-virtual-output: Check for DRI3 more carefully

Using xcb, we cannot simply call xcb_dri3_query_version() without it
terminating the connection if DRI3 is not enabled on the target display.
Oops.

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=89172
Signed-off-by: Chris Wilson &lt;chris&#64;chris-wilson.co.uk&gt;

diff --git a/tools/virtual.c b/tools/virtual.c
index 8e2b4a2..3bcd057 100644
<span class="hl kwb">--- a/tools/virtual.c</span>
<span class="hl kwa">+++ b/tools/virtual.c</span>
&#64;&#64; -218,6 +218,13 &#64;&#64; static inline XRRScreenResources *_XRRGetScreenResourcesCurrent(Display *dpy, Wi
 static int _x_error_occurred;
 
 static int
<span class="hl kwa">+_io_error_handler(Display *display)</span>
<span class="hl kwa">+{</span>
<span class="hl kwa">+	fprintf(stderr, &quot;XIO error on display %s\n&quot;, DisplayString(display));</span>
<span class="hl kwa">+	abort();</span>
<span class="hl kwa">+}</span>
<span class="hl kwa">+</span>
<span class="hl kwa">+static int</span>
 _check_error_handler(Display     *display,
 		     XErrorEvent *event)
 {
&#64;&#64; -320,6 +327,7 &#64;&#64; can_use_shm(Display *dpy,
 #include &lt;X11/Xlib-xcb.h&gt;
 #include &lt;X11/xshmfence.h&gt;
 #include &lt;xcb/xcb.h&gt;
<span class="hl kwa">+#include &lt;xcb/xcbext.h&gt;</span>
 #include &lt;xcb/dri3.h&gt;
 #include &lt;xcb/sync.h&gt;
 static Pixmap dri3_create_pixmap(Display *dpy,
&#64;&#64; -357,6 +365,7 &#64;&#64; static int dri3_query_version(Display *dpy, int *major, int *minor)
 {
 	xcb_connection_t *c = XGetXCBConnection(dpy);
 	xcb_dri3_query_version_reply_t *reply;
<span class="hl kwa">+	xcb_generic_error_t *error;</span>
 
 	*major = *minor = -1;
 
&#64;&#64; -364,7 +373,8 &#64;&#64; static int dri3_query_version(Display *dpy, int *major, int *minor)
 					     xcb_dri3_query_version(c,
 								    XCB_DRI3_MAJOR_VERSION,
 								    XCB_DRI3_MINOR_VERSION),
<span class="hl kwb">-					     NULL);</span>
<span class="hl kwa">+					     &amp;error);</span>
<span class="hl kwa">+	free(error);</span>
 	if (reply == NULL)
 		return -1;
 
&#64;&#64; -377,8 +387,14 &#64;&#64; static int dri3_query_version(Display *dpy, int *major, int *minor)
 
 static int dri3_exists(Display *dpy)
 {
<span class="hl kwa">+	xcb_extension_t dri3 = { &quot;DRI3&quot;, 0 };</span>
<span class="hl kwa">+	const xcb_query_extension_reply_t *ext;</span>
 	int major, minor;
 
<span class="hl kwa">+	ext = xcb_get_extension_data(XGetXCBConnection(dpy), &amp;dri3);</span>
<span class="hl kwa">+	if (ext == NULL || !ext-&gt;present)</span>
<span class="hl kwa">+		return 0;</span>
<span class="hl kwa">+</span>
 	if (dri3_query_version(dpy, &amp;major, &amp;minor) &lt; 0)
 		return 0;
 
&#64;&#64; -3228,6 +3244,7 &#64;&#64; int main(int argc, char **argv)
 		return -ret;
 
 	XSetErrorHandler(_check_error_handler);
<span class="hl kwa">+	XSetIOErrorHandler(_io_error_handler);</span>
 
 	ret = add_fd(&amp;ctx, display_open(&amp;ctx, src_name));
 	if (ret) {
<span class="hl kwb">-- </span>
cgit v0.10.2

</code></pre></td></tr></table>
</div> <!-- class=content -->
<div class="foot" style="padding-left:1em;padding-right:1em;">
<p>Copyright &copy; 2002-2014 <a href="mailto:jvinet@zeroflux.org"
title="contact Judd Vinet">Judd Vinet</a> and <a href="mailto:aaron@archlinux.org"
title="contact Aaron Griffin">Aaron Griffin</a>. The Arch Linux name and logo
are recognized trademarks. Some rights reserved. The registered trademark
Linux&reg; is used pursuant to a sublicense from LMI, the exclusive licensee
of Linus Torvalds, owner of the mark on a world-wide basis.</p>
</div>
</div> <!-- id=cgit -->
</body>
</html>
